cpp/src/arrow/extension_type.cc

```diff
@@ -144,13 +145,16 @@ namespace internal {
 static void CreateGlobalRegistry() {
   g_registry = std::make_shared<ExtensionTypeRegistryImpl>();
 
+  std::vector<std::shared_ptr<DataType>> ext_types{::arrow::extension::json()};
 #ifdef ARROW_JSON
-  // Register canonical extension types
-  auto ext_type =
-      checked_pointer_cast<ExtensionType>(extension::fixed_shape_tensor(int64(), {}));
-
-  ARROW_CHECK_OK(g_registry->RegisterType(ext_type));
+  ext_types.push_back(extension::fixed_shape_tensor(int64(), {}));
```

Comment:
Is the placement of `json()` and `fixed_shape_tensor()` correct?

Reply:
As in alphabetical order? Changed to:
```cpp
#ifdef ARROW_JSON
  std::vector<std::shared_ptr<DataType>> ext_types{
      extension::fixed_shape_tensor(int64(), {}), ::arrow::extension::json()};
#else
  std::vector<std::shared_ptr<DataType>> ext_types{::arrow::extension::json()};
#endif
```